<?xml version="1.0"?>
<launch>
    <arg name="camera" default="false" /> 
    <arg name="ns" default=''/> 

    <!-- Load robot config file -->
    <rosparam command="load" file="$(find rosbot_bringup)/config/rosbot_config.yaml" />
    <rosparam command="load" file="$(find emc_system)/config/rosbot_mrc_config.yaml" />

    <param name="robot_description" command="$(find xacro)/xacro '$(find rosbot_description)/urdf/rosbot.xacro'"/>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/>
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"/>

     <!-- Publish frame between base_link and laser depnding on whether namespace has been defined -->
    <group if= "$(eval arg('ns') != '')">
        <node pkg="tf" type="static_transform_publisher" name="ROSbot_laser" args="0 0 0 3.14 0 0 $(arg ns)/base_link $(arg ns)/laser 100" />
    </group>

    <group if= "$(eval arg('ns') == '')">
        <node pkg="tf" type="static_transform_publisher" name="ROSbot_laser" args="0 0 0 3.14 0 0 base_link laser 100" />
    </group>    

    <!-- Log Battery Voltage-->
    <node pkg="rosbot_bringup" type="battery_logger_node" name="battery_logger"/>

    <!-- Robot version is an enviroment variable-->
    <arg name="robot_version" default="$(env ROSBOT_VER)" />

    <!-- Run ROSBOT pro launch files-->
    <group if= "$(eval arg('robot_version') == 'ROSBOT_2.0_PRO')">      
        <include file="$(find rosbot_bringup)/launch/startlidar.launch">
            <arg name="rosbot_pro" value="true" />
            <arg name="ns" value="$(arg ns)"/>
        </include>
        <include file="$(find rosbot_ekf_multi)/launch/all.launch">
            <arg name="rosbot_pro" value="true" />
            <arg name="ns" value="$(arg ns)"/>
        </include>
    </group>
    
    <!-- Run ROSBOT non-pro launch files-->
    <group if= "$(eval arg('robot_version') != 'ROSBOT_2.0_PRO')">
        <include file="$(find rosbot_bringup)/launch/startlidar.launch">
            <arg name="rosbot_pro" value="false" />
            <arg name="ns" value="$(arg ns)"/>
        </include>
        <include file="$(find rosbot_ekf_multi)/launch/all.launch">
            <arg name="rosbot_pro" value="false" />
            <arg name="ns" value="$(arg ns)"/>
        </include>
    </group>    

    <!-- Run camera launch files-->
    <group if= "$(eval camera)">
        <include file="$(find astra_launch)/launch/astra.launch">
            <arg unless= "$(eval arg('ns') == '')" name="camera" value="$(arg ns)_camera"/>
            <arg if= "$(eval arg('ns') == '')" name="camera" value="camera"/>
        </include>

        <!-- Publish Camera -> base_link transform-->
        <node pkg="tf" type="static_transform_publisher" name="camera_publisher" args="-0.03 0 0.18 0 0 0  $(arg ns)/base_link $(arg ns)_camera_link 100" />
    </group>

    <include file="$(find rosbot_bringup)/launch/laser_transformer.launch"/>
    <node pkg="rosbot_bringup" type="bumper_emulator_node" name="bumper_emulator"/>
    
</launch>
